clients/src/main/resources/common/message/MetadataSnapshotFooterRecord.json

```diff
@@ -0,0 +1,25 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+{
+  "type": "data",
+  "name": "MetadataSnapshotFooterRecord",
```

Comment:
This is generic to snapshots, rather than metadata snapshots, right? In other words, if we have another snapshotted topic, we'd want to use this record as well.

So I don't think it should have "metadata" in the name

Reply:
Good Point. I had initially made this to be a generic Header and Footer record, but then scoped it to metadata. I did not know we had the intention of expanding the snapshot concept to other topics. (IIUC snapshots would only work on topics that are delta logged, and no other topics work that way today). But if we will do that, then I agree with you.

I guess my only counter question would be that given today the metadata topic is kind of special,  do you envision the metadata snapshot schema evolving different (richer in information) from other snapshot-able topics, such that it might need it's own header/footer?

Reply:
Perhaps. However, if we needed a specific metadata header or footer, we'd do that as a metadata record.